src/pages/ProductsPage/ProductsPage.tsx

```diff
@@ -0,0 +1,37 @@
+import { Box } from '@mui/material';
+import Button from '@mui/material/Button';
+import { GridToolbar} from '@mui/x-data-grid/components/toolbar';
+import { DataGrid } from '@mui/x-data-grid/DataGrid';
+import { productTableColumns } from '../../constants/productTableColumns';
+import { ProductModel } from '../../models/ProductModel';
+
+interface Props {
+  getProductsClick: () => void;
+  products: { [productId: number]: ProductModel };
```

Comment:
You have to rework these props: create just an array to avoid the Object.values(props.products) call. It does your ProductPage component a little complicated. The component should accept what it needs and just render. You should map the ProductModel[]  in the container using an array of ids, for example:
`const products: ProductModel[] = useAppSelector((store) =>
    store.catalog.productsIds.map((id) => store.catalog.products[id])
  );`
and then pass the products into the component